### ISSUE TITLE
Trouble finding trilinos

### ISSUE THREAD

**Issue by VictorEijkhout** [dealii/dealii#14812]
I'm not sure what broke in my setup.....

```
cmake -D CMAKE_INSTALL_PREFIX=/work2/00434/eijkhout/dealii/installation-git-clx-intel-impi -D DEAL_II_WITH_MPI=ON -D DEAL_II_CXX_FLAGS=-march=native -D DEAL_II_COMPONENT_EXAMPLES=OFF -D DEAL_II_WITH_P4EST=ON    -D P4EST_DIR=/work2/00434/eijkhout/p4est/installation-2.8-clx-intel-impi -D DEAL_II_WITH_PETSC=ON    -D PETSC_DIR=/work2/00434/eijkhout/petsc/petsc-3.18.3 -D DEAL_II_WITH_TRILINOS=ON -D TRILINOS_DIR=/work2/00434/eijkhout/trilinos/installation-git-clx-intel-impi                     -D CMAKE_VERBOSE_MAKEFILE=ON                     /work2/00434/eijkhout/dealii/dealii-git
```

with 

```
[c202-001 dealii:13] ls /work2/00434/eijkhout/trilinos/installation-git-clx-intel-impi
bin/  include/  lib/  TrilinosRepoVersion.txt
[c202-001 dealii:14] find !$ -name \*.cmake | wc -l
find /work2/00434/eijkhout/trilinos/installation-git-clx-intel-impi -name \*.cmake | wc -l
527
[c202-001 dealii:16] ls /work2/00434/eijkhout/trilinos/installation-git-clx-intel-impi/lib/cmake/Stokhos/
StokhosConfig.cmake  StokhosTargets.cmake  StokhosTargets-release.cmake
```

but:

```
-- Include /work2/00434/eijkhout/dealii/dealii-git/cmake/configure/configure_20_trilinos.cmake
-- Enabled Kokkos devices: SERIAL
-- Enabled Kokkos devices: SERIAL
-- Found EPETRA_CONFIG_H
-- TRILINOS_LIBRARY_Panzer::all_libs not found! The call was:
--     find_library(TRILINOS_LIBRARY_Panzer::all_libs NAMES Panzer::all_libs HINTS NO_DEFAULT_PATH NO_CMAKE_ENVIRONMENT_PATH NO_CMAKE_PATH NO_SYSTEM_ENVIRONMENT_PATH NO_CMAKE_SYSTEM_PATH NO_CMAKE_FIND_ROOT_PATH)
-- TRILINOS_LIBRARY_PanzerDiscFE::all_libs not found! The call was:
--     find_library(TRILINOS_LIBRARY_PanzerDiscFE::all_libs NAMES PanzerDiscFE::all_libs HINTS NO_DEFAULT_PATH NO_CMAKE_ENVIRONMENT_PATH NO_CMAKE_PATH NO_SYSTEM_ENVIRONMENT_PATH NO_CMAKE_SYSTEM_PATH NO_CMAKE_FIND_ROOT_PATH)
-- TRILINOS_LIBRARY_PanzerDofMgr::all_libs not found! The call was:
--     find_library(TRILINOS_LIBRARY_PanzerDofMgr::all_libs NAMES PanzerDofMgr::all_libs HINTS NO_DEFAULT_PATH NO_CMAKE_ENVIRONMENT_PATH NO_CMAKE_PATH NO_SYSTEM_ENVIRONMENT_PATH NO_CMAKE_SYSTEM_PATH NO_CMAKE_FIND_ROOT_PATH)
-- TRILINOS_LIBRARY_PanzerCore::all_libs not found! The call was:
--     find_library(TRILINOS_LIBRARY_PanzerCore::all_libs NAMES PanzerCore::all_libs HINTS NO_DEFAULT_PATH NO_CMAKE_ENVIRONMENT_PATH NO_CMAKE_PATH NO_SYSTEM_ENVIRONMENT_PATH NO_CMAKE_SYSTEM_PATH NO_CMAKE_FIND_ROOT_PATH)
-- TRILINOS_LIBRARY_Piro::all_libs not found! The call was:
--     find_library(TRILINOS_LIBRARY_Piro::all_libs NAMES Piro::all_libs HINTS NO_DEFAULT_PATH NO_CMAKE_ENVIRONMENT_PATH NO_CMAKE_PATH NO_SYSTEM_ENVIRONMENT_PATH NO_CMAKE_SYSTEM_PATH NO_CMAKE_FIND_ROOT_PATH)
-- TRILINOS_LIBRARY_Stokhos::all_libs not found! The call was:
--     find_library(TRILINOS_LIBRARY_Stokhos::all_libs NAMES Stokhos::all_libs HINTS NO_DEFAULT_PATH NO_CMAKE_ENVIRONMENT_PATH NO_CMAKE_PATH NO_SYSTEM_ENVIRONMENT_PATH NO_CMAKE_SYSTEM_PATH NO_CMAKE_FIND_ROOT_PATH)
```

et cetera.

**Comment by VictorEijkhout** [comment]
I could use some help with this.....

**Comment by masterleinad** [comment]
It's hard to help with such little information. What Trilinos version are you using? Which Trilinos packages do you actually need? Can any of the required Trilinos packages not be found or are only optional ones missing?
You were saying something broke in your setup implying that it was working before. What did you change?

**Comment by VictorEijkhout** [comment]
I'm using the latest repo pull

Trilinos package: I'm not using anything. You are.

I have (as documented in my question above)

- set TRILINOS_DIR on the commandline
- set CMAKE_MODULE/PREFIX_DIR

and all those seem to exist. 

I've been rewriting all my makefile, and I've switched to the latest versions. Hard to tell what the breaking step is.

**Comment by VictorEijkhout** [comment]
Would the cmake log file be of any use?

**Comment by tamiko** [comment]
@VictorEijkhout What Trilinos version is this?

This might be related to a recent change in Trilinos - they switched to import targets in their current development version. The fact that we mistakenly try to find the library `TRILINOS_LIBRARY_Panzer::all_libs` strongly suggests that.

**Comment by VictorEijkhout** [comment]
I'm using a recent pull from the trilinos repo. The latest release 13-4-1 does not compile at all with the Intel compiler.

**Comment by masterleinad** [comment]
> I'm using a recent pull from the trilinos repo. The latest release 13-4-1 does not compile at all with the Intel compiler.

We don't support Trilinos later than their latest release yet. 
Out of curiosity: Is the "Intel compiler" icpc (Intel Classic Compiler) or icpx (Intel oneAPI Compiler)? I would always go for the latter if I have the choice.

**Comment by VictorEijkhout** [comment]
This problem seems to have been solved in your 9.5.1 release.
